File: DSA/patterns/multiple_pointers/unique_count.py
#!/bin/python3

'''
    Problem:
    Given a sorted array, count the number of unique values.
'''

def count_unique_values(arr):
    """
        Problem Solution.
    """
    if not arr:
        return 0

    left = 0

    for right in range(1, len(arr)):
        if arr[left] != arr[right]:
            left += 1
            arr[left] = arr[right]

    return left + 1

# Example Usage
print(count_unique_values([1, 1, 2, 2, 3, 4, 4, 5]))  # Output: 5
print(count_unique_values([]))                        # Output: 0

File: DSA/patterns/multiple_pointers/uniqueCount.js
function countUniqueValues(arr) {
  if (arr.length === 0) return 0;

  let left = 0;

  for (let right = 1; right < arr.length; right++) {
    if (arr[left] !== arr[right]) {
      left++;
      arr[left] = arr[right];
    }
  }

  return left + 1;
}

// Example Usage
console.log(countUniqueValues([1, 1, 2, 2, 3, 4, 4, 5])); // Output: 5
console.log(countUniqueValues([])); // Output: 0
